tests/sort-requires.js

```diff
@@ -0,0 +1,623 @@
+const rule = require('../lib').rules['sort-requires'];
+const { RuleTester } = require('eslint');
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2020
+  }
+});
+
+ruleTester.run('sort', rule, {
+  invalid: [
+    // Declaration sort - default options
+    {
+      code: `
+        const a = require('foo');
+        const A = require('foo');
+        const c = require('foo');
+      `,
+      errors: [{ message: "Expected require declaration of 'A' to be placed before 'a'.", type: 'VariableDeclaration' }]
+    },
+    {
+      code: `
+        const b = require('foo');
+        const a = require('foo');
+      `,
+      errors: [{ message: "Expected require declaration of 'a' to be placed before 'b'.", type: 'VariableDeclaration' }]
+    },
+    {
+      code: `
+        const a = require('foo');
+        require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'none' syntax to be placed before 'single' syntax.",
+          type: 'ExpressionStatement'
+        }
+      ]
+    },
+    {
+      code: `
+        const c = require('foo');
+        const { a, b } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ]
+    },
+    {
+      code: `
+        const c = require('foo');
+        const { d, e } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ]
+    },
+    {
+      code: `
+        const { ...x } = require('foo');
+        const { y, ...z } = require('foo');
+      `,
+      errors: [
+        {
+          message: "Expected require declaration of 'multiple' syntax to be placed before 'single' syntax.",
+          type: 'VariableDeclaration'
+        }
+      ]
+    },
+    // Declaration sort - default options with `autofixDeclarationSortOrder`
+    {
+      code: `
+        const { d, f } = require('foo');
+        const { c, e } = require('foo');
+        const a = require('foo');
+        const { b } = require('foo');
+        const A = require('foo');
+        const { ...Z } = require('foo');
+      `,
+      errors: [
+        { message: "Expected require declaration of 'c' to be placed before 'd'.", type: 'VariableDeclaration' },
+        { message: "Expected require declaration of 'A' to be placed before 'b'.", type: 'VariableDeclaration' }
+      ],
+      options: [{ autofixDeclarationSortOrder: true }],
+      output: `
+        const { c, e } = require('foo');
+        const { d, f } = require('foo');
+        const A = require('foo');
+        const { ...Z } = require('foo');
+        const a = require('foo');
+        const { b } = require('foo');
```

Comment:
I would argue that this require should be the first one, it is using the multiple syntax even if just declaring one.

Reply:
That's the case when `useOldSingleMemberSyntax` option is on. We will turn it on in our `eslint-config-uphold`.